git-cliff/src/lib.rs

```diff
@@ -100,9 +105,9 @@ pub fn run(mut args: Opt) -> Result<()> {
 	if args.body.is_some() {
 		config.changelog.body = args.body;
 	}
-	if args.sort == "oldest" {
+	if args.sort == Sort::Oldest {
 		if let Some(ref sort_commits) = config.git.sort_commits {
-			args.sort = sort_commits.to_string();
+			args.sort = Sort::from_str(sort_commits, false).expect("");
```

Comment:
I think it is safe to ignore case here. (`from_str`)

Also, instead of using expect without a message, I'd suggest using `if-let` or implementing `Default` for `Sort` and unwrapping that way. Like so:

```rs

impl Default for Sort {
	fn default() -> Self {
		Self::Oldest
	}
}

args.sort = Sort::from_str(sort_commits, true).unwrap_or_default();
```

Or something simpler can be applied too. But not empty `expect` 🙂

Can you please make this change? 😊

Reply:
I'm wondering, if this is parsed from a config file, then doing a `unwrap_or_default` silently ignores incorrect values, is that desirable?

Reply:
Hmm, that's a good point. It'd be best if we implement `De/Serialize` on `Sort` and use it in `GitConfig` really 😄 Then in this case you can use `expect` with a message at least 🙈 I think it's not possible to hit this error but still, for the sake of _code quality_...

Reply:
In my experience, making a common enum for both config files and clap flags creates coupling issues between the cli and config crates (here `git_cliff` and `git_cliff_core`. Instead, I'd rather create a separate `Sort` enum for `git_cliff_core` and then `impl From<git_cliff_core::Sort> for Sort`

Reply:
Instead let's keep it simple and give this expect a message :)